### PR TITLE
Updated filtering for vulnerabilities table

### PR DESCRIPTION
**Pull request by DrAlexD**
### What's done:
- removed useless filter `isOwner` in `VulnerabilityFilter`.
- added filtering by `Summary` to field for `Identifier` filtering, field renamed to `Identifier or Summary`.
- added field for filtering by `Status`.
- for better understanding, the displayed status description changed for `AUTO_APPROVED` to `Approved` (as for `APPROVED`) to display `AUTO_APPROVED` and `APPROVED` statuses together, and for `PENDING_REVIEW` to `Edited`.

gif15

### CODE REVIEW

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt:129]: it's not a prefix anymore, also we can search by description too and suggest to rename it to `freeText` (or something like it)

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt
Comment (nulls) [inline save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt:18]: why uploaded?

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt
Comment (DrAlexD) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt:129]: I think that it may not be clear to user if search result shows vulnerabilities that we found by `Details`, because this field isn't displayed in the table.

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt
Comment (DrAlexD) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt:18]: Only vulnerabilities uploaded using .json files get `AUTO_APPROVED` status, while they don't go through `Created` stage and do not require to be `APPROVED` from admin. Therefore, for better understanding, I decided to describe such vulnerabilities with `UPLOADED` status.

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt
Comment (DrAlexD) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt:147]: I thought about this, but in the end I decided to highlight the difference between these two statuses by adding the description `Uploaded` for `AUTO_APPROVED`. I thought that it might not be obvious to user why some of the vulnerabilities have `Created` status after creation and require approval, while the other part loaded using files is immediately in the `APPROVED` status (now in `AUTO_APPROVED`).

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt
Comment (nulls) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt:18]: I think better to leave auto approved. @akuleshov7

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt
Comment (nulls) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt:129]: agree, but still suggest to rename to `freeText`

Comment (DrAlexD) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt:129]: Change `freeText` for property in `VulnerabilityFilter` and `FreeText...` for input field? Will it be clear to user that search is performed by `Identifier` and `Summary`? Or only `freeText` for property?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt
Comment (DrAlexD) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt:578]: Because in past commits I accidentally changed `values()` ​​to `entries` based on a suggestion from IDEA, without thinking about the fact that `entries` are now immutable and in some cases the logic may be broken. Therefore, more thorough analysis is required, but for now I decided to return `values()`.

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt
Comment (nulls) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt:129]: discussed offline

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt
Comment (DrAlexD) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt]: Done. These statuses are now displayed together under a common description `Approved`.

**Review by akuleshov7 — APPROVED** [review]
Super! Thanks

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt
Comment (akuleshov7) [inline save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt:18]: Strange that diktat is not triggering here (APPROVED actually should go first)

Comment (DrAlexD) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt:18]: Diktat triggered here, I added `@Suppress` for this, this order seemed more logical to me. I can bring the old order back if needed.

**Review by nulls — APPROVED** [review]

**Review by Cheshiriks — APPROVED** [review]

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt
Comment (DrAlexD) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityStatus.kt]: Decided to return old order

**Pull request merged**
